dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Inlining.java

```diff
@@ -71,7 +71,16 @@ private boolean canInline(FunctionCall call) {
     }
 
     private void inlineAllCalls(Function function, Map<Function, Snapshot> snapshots) {
-        int scopeCounter = 0;
+        int scope = 0;
+        // Advance scope counter.
+        for (Register r : function.getRegisters()) {
+            final int i = r.getName().indexOf(":");
+            if (i == -1) { continue; }
+            try {
+                scope = Integer.max(scope, Integer.parseInt(r.getName().substring(0, i)));
```

Comment:
Why do we need all this?

Reply:
I would also refactor this code into a helper function `getNextFreeScopeId`